backend/bitboxbase/rpcclient/rpcclient.go

```diff
@@ -0,0 +1,174 @@
+	"github.com/digitalbitbox/bitbox-wallet-app/util/logging"
+	"github.com/digitalbitbox/bitbox-wallet-app/util/observable"
+	"github.com/digitalbitbox/bitbox-wallet-app/util/observable/action"
+	"github.com/flynn/noise"
+	"github.com/gorilla/websocket"
+
+	"github.com/sirupsen/logrus"
+)
+
+type rpcConn struct {
+	readChan  <-chan []byte
+	writeChan chan<- []byte
+	quitChan  chan<- struct{}
+}
+
+func (conn *rpcConn) Read(p []byte) (n int, err error) {
+	message := <-conn.readChan
+	return copy(p, message), nil
+}
+
+func (conn *rpcConn) Write(p []byte) (n int, err error) {
+	conn.writeChan <- p
+	return len(p), nil
+}
+
+func (conn *rpcConn) Close() error {
+	close(conn.quitChan)
+	return nil
+}
+
+// GetEnvResponse holds the information from the rpc call reply to get some environment data from the base
+type GetEnvResponse struct {
+	Network        string
+	ElectrsRPCPort string
+}
+
+// SampleInfoResponse holds some sample information from the BitBox Base
+type SampleInfoResponse struct {
+	Blocks         int64   `json:"blocks"`
+	Difficulty     float64 `json:"difficulty"`
+	LightningAlias string  `json:"lightningAlias"`
+}
+
+// RPCClient implements observable blockchainInfo.
+type RPCClient struct {
+	observable.Implementation
+	sampleInfo          *SampleInfoResponse
+	log                 *logrus.Entry
+	address             string
+	bitboxBaseConfigDir string
+
+	bitboxBaseNoiseStaticPubkey   []byte
+	channelHash                   string
+	channelHashAppVerified        bool
+	channelHashBitBoxBaseVerified bool
+	sendCipher, receiveCipher     *noise.CipherState
+
+	weHaveQuitChan chan<- struct{}
+
+	//rpc stuff
+	client       *rpc.Client
+	bitboxBaseID string
+}
+
+// NewRPCClient returns a new bitboxbase rpcClient.
+func NewRPCClient(address string, bitboxBaseConfigDir string) *RPCClient {
+	rpcClient := &RPCClient{
+		log:                 logging.Get().WithGroup("bitboxbase"),
+		address:             address,
+		sampleInfo:          &SampleInfoResponse{},
+		bitboxBaseConfigDir: bitboxBaseConfigDir,
+	}
+	return rpcClient
+}
+
+// Connect starts the websocket go routine, first checking if the middleware is reachable,
+// then establishing a websocket connection, then authenticating and encrypting all further traffic with noise.
+func (rpcClient *RPCClient) Connect(address string, bitboxBaseID string) error { //, bitboxBaseEvent chan proto.Message, apiMap map[string]chan *basemessages.BitBoxBaseOut) error {
+	rpcClient.bitboxBaseID = bitboxBaseID
+	response, err := http.Get("http://" + address + "/")
+	if err != nil {
+		rpcClient.log.Println("No response from middleware", err)
+		return err
+	}
+
+	if response.StatusCode != http.StatusOK {
+		rpcClient.log.Println("Received http status code from middleware other than 200")
+		return err
+	}
+
+	bodyBytes, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		rpcClient.log.Println("Body Bytes not read properly")
+		return err
+	}
+	_, err = regexp.MatchString("OK!", string(bodyBytes))
```

Comment:
not needed imho, matching text with regexp is often not robust, and you already check the status code above, which is what really matters.